src/JSInterop/Microsoft.JSInterop/src/JSRuntime.cs

```diff
@@ -174,6 +183,46 @@ protected virtual void BeginInvokeJS(long taskId, string identifier, string? arg
             DotNetInvocationInfo invocationInfo,
             in DotNetInvocationResult invocationResult);
 
+        /// <summary>
+        /// Invoked by JS
+        /// </summary>
+        /// <param name="id">Atomically incrementing identifier for the byte array being transfered.</param>
+        /// <param name="data">Byte array to be transfered to JS.</param>
+        protected internal virtual void SendByteArray(int id, byte[] data)
+        {
+            throw new NotSupportedException("JSRuntime subclasses are responsible for implementing byte array transfer to JS.");
+        }
+
+        /// <summary>
+        /// Accepts the byte array data being transferred from JS to DotNet.
+        /// </summary>
+        /// <param name="id">Identifier for the byte array being transfered.</param>
+        /// <param name="data">Byte array to be transfered from JS.</param>
+        public void ReceiveByteArray(int id, byte[] data)
+        {
+            if (id == 0)
+            {
+                // Starting a new transfer, clear out previously stored byte arrays
+                ResetByteArraysToBeRevived();
+            }
+            else if (id != (ByteArraysToBeRevived.Count + 1))
+            {
+                throw new ArgumentOutOfRangeException($"Element id '{id}' cannot be added to the byte arrays to be revived with length '{ByteArraysToBeRevived.Count}'.");
+            }
+            else if (SignalRMaxMessageSizeBytes - data.Length < ByteArraysToBeRevivedTotalBytes)
```

Comment:
I find more natural this check to be expressed as `ByteArraysToBeRevivedTotalBytes + data.Length > SignalRMessageSizeInBytes`. SignalRMessageSizeInBytes is a static limit and the other two things are the ones that change.

Reply:
The concern in previous review comments was about possible numeric overflow, hence the subtraction rather than addition.

TBH that concern seems rather artificial to me if it's a `long` (you'd have to send 8 exabytes, and as we know, nobody will even need more than 640KB).

Reply:
If we are concerned about overflow we can use `checked()`. I didn't realize there was a previous comment about potential overflows and we changed it as a result.

I'm fine if we leave it as it is, I don't think it's a super powerful reason to have it done that way at the cost of readability for what you just described, however I'm fine if we don't change it.

Reply:
It wasn't me who was concerned about overflows here before, and I'm still not concerned about it 😄

Reply:
The runtime tends to use this pattern (e.g. https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/IO/TextWriter.cs#L173), so it's not really that unusual. Could we let this be?

Reply:
As I said, I'm fine with it, I just haven't seen it in our code base and took me 5 seconds to get my head around it.